### ISSUE TITLE
Implement firmware flashing on Mac

### ISSUE THREAD

**Issue by chris007de** [AllYarnsAreBeautiful/ayab-desktop#157]
Currently, firmware flashing does not work on Mac systems, simply because it is not implemented. Somebody who owns a Mac or does have access to one should check what's the best way to deliver/assume avrdude and implement the correct avrdude call in firmware_flash.py

**Comment by chris007de** [comment]
@oskay Do you have the possibility to work on this?

**Comment by oskay** [comment]
I do have experience flashing AVR firmware from the mac (but not in python). I will take a look.

**Comment by oskay** [comment]
The best way to implement this (that is, including the avrdude) will depend upon #158 -- exactly how everything is packaged.

There is precedent for this kind of thing, even on the Mac. The Arduino application for the Mac includes a binary distribution of the AVR toolset, including avrdude. (I will note that the avr tools directory (normally found at `/Applications/Arduino.app/Contents/Java/hardware/tools/avr`) is a hefty 147 MB. Most of this is gcc, however, and I expect that paring out the unneeded parts would bring it down to a justifiable size.)

Right now the Mac version of AYAB desktop does not have the Load AYAB Firmware menu option implemented-- I'm not sure if this is an intentional choice or if the window is simply not loading due to an error. Is there a location in the source code that this is intentionally switched?

**Comment by oskay** [comment]
I have been able to get firmware flashing to work on the Mac. It appears to be working well now, and I'll be submitting a pull request shortly in the pyqt5 branch.

To summarize, I have made the following changes:
* Added a binary copy of avrdude for the mac, along with the relevant libusb library files.
** Both avrdude and libusb are already bundled with AYAB, so there shouldn't be any significant consequences (for example, with respect to licensing) needed
* Updated avrdude.conf to a newer version. The programmer definitions in the old version (2011) were apparently out of date.
* Added a case to detect MacOS, alongside the windows

The avrdude binary and avrdude.conf were copied from a recent Mac release of Arduino.  This is avrdude version 6.0.1. (Not quite up to date, but closer....)

One concern: Updating avrdude.conf  _may_ possibly require that the avrdude binaries for windows and/or linux should be updated to newer versions as well. (Even if it is not absolutely required, it may be a good idea.)
